core/src/main/scala/org/ensime/indexer/domain.scala

```diff
@@ -47,27 +47,34 @@ case class ClassName(pack: PackageName, name: String) extends FullyQualifiedName
     if (pack.path.isEmpty) name
     else ClassName.cleanupPackage(pack.fqnString + "." + name)
 
-  def internalString =
-    "L" + (if (pack.path.isEmpty) name else pack.path.mkString("/") + "/" + name) + ";"
+  def internalString = name match {
```

Comment:
The `ClassName` constructor and 3 parameter apply would still be exposed (can't be overloaded).

I found a way to split it, in a safe manner. I've pushed updates, have a look.